mmcls/models/backbones/convnext.py

```diff
@@ -0,0 +1,311 @@
+
+    Args:
+        num_channels (int): The number of channels of the input tensor.
+        dim (int): The dimension of channel in the input tensor. Defaults to 1.
+        eps (float): a value added to the denominator for numerical stability.
+            Defaults to 1e-6.
+        elementwise_affine (bool): a boolean value that when set to ``True``,
+            this module has learnable per-element affine parameters initialized
+            to ones (for weights) and zeros (for biases). Defaults to True.
+
+    Note:
+        Comparing with the original implementation, this implementation uses
+        ``dim`` instead of ``data_format`` to specify the dimension of channel.
+
+        1. For inputs with shape (batch_size, height, width, channels), use
+           ``dim=-1`` or ``dim=3``.
+        2. For inputs with shape (batch_size, channels, height, width), use
+           ``dim=1``.
+    """
+
+    def __init__(self,
+                 num_channels: int,
+                 dim: int = 1,
+                 eps: float = 1e-6,
+                 **kwargs) -> None:
+        super().__init__(num_channels, eps=eps, **kwargs)
+        self.num_channels = self.normalized_shape[0]
+        self.dim = dim
+
+    def forward(self, x):
+        if self.dim == -1 or self.dim == x.dim() - 1:
+            return F.layer_norm(x, self.normalized_shape, self.weight,
+                                self.bias, self.eps)
+        else:
+            u = x.mean(self.dim, keepdim=True)
+            s = (x - u).pow(2).mean(self.dim, keepdim=True)
+            x = (x - u) / torch.sqrt(s + self.eps)
+
+            weight_shape = [1] * self.dim + [self.num_channels] + [1] * (
+                x.dim() - self.dim - 1)
+            x = self.weight.view(weight_shape) * x + self.bias.view(
+                weight_shape)
+            return x
+
+
+class ConvNeXtBlock(BaseModule):
+    """ConvNeXt Block.
+
+    Args:
+        dim (int): Number of input channels.
+        drop_path (float): Stochastic depth rate. Default: 0.0
+        channels_last (bool): Whether to permute the channel's dim to the last
+            before the layer norm.More details can be found in the note.
+            Defaults to True.
+        layer_scale_init_value (float): Init value for Layer Scale.
+            Defaults to 1e-6.
+
+    Note:
+        There are two equivalent implementations:
+
+        1. DwConv -> LayerNorm (channels_first) -> 1x1 Conv -> GELU
+           -> 1x1 Conv; all outputs are in (N, C, H, W).
+        2. DwConv -> Permute to (N, H, W, C) -> LayerNorm (channels_last)
+           -> Linear -> GELU -> Linear; Permute back
+
+        As default, we use the second to align with the official repository.
+        And it may be slightly faster because LayerNorm on the last dim is
+        officially supported by PyTorch.
+    """
+
+    def __init__(self,
+                 in_channels,
+                 drop_path_rate=0.,
+                 channels_last=True,
```

Comment:
`channels_last` may make users confused, users may see it as letting inputs as shape (N, H, W, C),  channel-dim in the last, but actually, It has nothing to do with the input tensor.

maybe replace it with  `use_pwconv(bool)=False` as timm